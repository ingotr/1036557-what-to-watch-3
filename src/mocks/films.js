const films = [
  {
    id: `mock-011`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    imgSrc: `img/aviator.jpg`,
    previewSrc: `https://download.blender.org/peach/trailer/trailer_400p.ogg`,
  },
  {
    id: `mock-012`,
    title: `Bohemian Rhapsody`,
    imgSrc: `img/aviator.jpg`,
    previewSrc: `https://download.blender.org/peach/trailer/trailer_400p.ogg`,
  },
  {
    id: `mock-013`,
    title: `Macbeth`,
    imgSrc: `img/aviator.jpg`,
    previewSrc: `https://download.blender.org/peach/trailer/trailer_400p.ogg`,
  },
  {
    id: `mock-014`,
    title: `Aviator`,
    imgSrc: `img/aviator.jpg`,
    previewSrc: `https://download.blender.org/peach/trailer/trailer_400p.ogg`,
  },
  {
    id: `mock-015`,
    title: `We need to talk about Kevin`,
    imgSrc: `img/aviator.jpg`,
    previewSrc: `https://download.blender.org/peach/trailer/trailer_400p.ogg`,
  },
  {
    id: `mock-016`,
    title: `What We Do in the Shadows`,
    imgSrc: `img/aviator.jpg`,
    previewSrc: `https://download.blender.org/peach/trailer/trailer_400p.ogg`,
  },
  {
    id: `mock-017`,
    title: `Revenant`,
    imgSrc: `img/aviator.jpg`,
    previewSrc: `https://download.blender.org/peach/trailer/trailer_400p.ogg`,
  },
  {
    id: `mock-018`,
    title: `Johnny English`,
    imgSrc: `img/aviator.jpg`,
    previewSrc: `https://download.blender.org/peach/trailer/trailer_400p.ogg`,
  },
];

const movieInfo = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  year: 2014,
  poster: {
    big: `img/the-grand-budapest-hotel-poster.jpg`,
    bigAlt: `The Grand Budapest Hotel poster`,
  },
  rating: {
    score: 8.9,
    level: `Very good`,
    count: 240,
  },
  director: `Wes Andreson`,
  description: [`In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave &#96s friend and protege.`,
    `Gustave prides himself on providing first-className service to the hotel &#96s guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave &#96s lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
};

export {films, movieInfo};
