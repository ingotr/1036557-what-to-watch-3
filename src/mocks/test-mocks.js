const FirstMovie = {
  TITLE: `Mock-test-The Grand Budapest Hotel`,
  GENRE: `Mock-test-Drama`,
  RELEASE_DATE: 2020,
};

const movies = [
  {
    id: `mock-test-011`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-012`,
    title: `Bohemian Rhapsody`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-013`,
    title: `Macbeth`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-014`,
    title: `Aviator`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-015`,
    title: `We need to talk about Kevin`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-016`,
    title: `What We Do in the Shadows`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-017`,
    title: `Revenant`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-018`,
    title: `Johnny English`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
];

const movieInfo = {
  title: `Mock-test-The Grand Budapest Hotel`,
  genre: `Mock-test-Drama`,
  year: 2017,
  poster: {
    big: `img/the-grand-budapest-hotel-poster.jpg`,
    bigAlt: `Mock-test-The Grand Budapest Hotel poster`,
  },
  rating: {
    score: 8.9,
    level: `Mocks-Very good`,
    count: 240,
  },
  director: `Mock-Wes Andreson`,
  description: [`Mock-test-In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave &#96s friend and protege.`,
    `Gustave prides himself on providing first-className service to the hotel &#96s guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave &#96s lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
  starring: [`Mock-test-Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
};

export {FirstMovie, movies, movieInfo};
