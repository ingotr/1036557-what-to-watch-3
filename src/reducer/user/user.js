import {extend} from '../../utils.js';

const SERVER_URL = `https://htmlacademy-react-3.appspot.com/`;

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  authErrorMessage: null,
  avatarUrl: ``,
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  SET_AVATAR_URL: `SET_AVATAR_URL`,
};

const ActionCreator = {
  requireAuthorization: (status) => {
    return {
      type: ActionType.REQUIRED_AUTHORIZATION,
      payload: status,
    };
  },
  setAvatarUrl: (avatarUrl) => {
    return {
      type: ActionType.SET_AVATAR_URL,
      payload: avatarUrl
    };
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload
      });
    case ActionType.SET_AVATAR_URL:
      return extend(state, {
        avatarUrl: `${SERVER_URL}${action.payload}`
      });
  }

  return state;
};

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then((response) => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
        dispatch(ActionCreator.setAvatarUrl(response.data.avatar_url));
      })
      .catch((err) => {
        throw err;
      });
  },

  login: (authData, onSuccess, onError) => (dispatch, getState, api) => {
    return api
      .post(`/login`, {
        email: authData.login,
        password: authData.password
      })
      .then((response) => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
        dispatch(ActionCreator.setAvatarUrl(response.data.avatar_url));
        onSuccess();
      })
      .catch((error) => {
        if (error.response) {
          onError(error.response.data.error);
        }
      });
  }
};


export {ActionCreator, ActionType, AuthorizationStatus,
  Operation, reducer,
};
