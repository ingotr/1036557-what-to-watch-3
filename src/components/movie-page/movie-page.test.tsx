import * as React from 'react';
import * as renderer from 'react-test-renderer';
import MoviePage from './movie-page';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import NameSpace from '../../reducer/name-space';
import {BrowserRouter} from 'react-router-dom';
import {AuthorizationStatus} from '../../reducer/user/user';
import testFunc from '../../utils';

import {movies, film} from '../../mocks/test-mocks';

const DEFAULT_GENRE = `All genres`;
const DEFAULT_MOVIES_COUNT = 8;

const mockStore = configureStore([]);

it(`Render App`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      currentGenre: DEFAULT_GENRE,
      moviesByGenre: movies.slice(0, DEFAULT_MOVIES_COUNT),
      showedMovies: movies,
      moviesCount: DEFAULT_MOVIES_COUNT,
    },
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.NO_AUTH,
      avatarUrl: ``
    }
  });

  const tree = renderer
    .create(
        <BrowserRouter>
          <Provider store={store}>
            <MoviePage
              authorizationStatus={AuthorizationStatus.NO_AUTH}
              avatarUrl=""
              movie={film}
              movies={movies}
              onMovieCardClick={testFunc}
              onFilmFavoriteStatusClick={testFunc}
            />
          </Provider>
        </BrowserRouter>, {
          createNodeMock: () => {
            return {};
          }
        })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
