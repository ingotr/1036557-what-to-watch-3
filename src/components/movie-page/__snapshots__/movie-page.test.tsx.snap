// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="The Grand Budapest Hotel"
          src="img/the-grand-budapest-hotel-bg.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            href="/login"
            onClick={[Function]}
          >
            Sign In
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            The Grand Budapest Hotel
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama
            </span>
            <span
              className="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/grand-budapest"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <a
              className="btn btn--list movie-card__button"
              href="/login"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </a>
            <a
              className="btn movie-card__button"
              href="/login"
              onClick={[Function]}
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="The Grand Budapest Hotel"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item "
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item "
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating movie-tab movie-tab--overview
        "
          >
            <div
              className="movie-rating__score"
            >
              8.9
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Very good
              </span>
              <span
                className="movie-rating__count"
              >
                240
                 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text movie-tab movie-tab--overview
        "
          >
            In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave &#96s friend and protege.
  Gustave prides himself on providing first-className service to the hotel &#96s
  guests, including satisfying the sexual needs of the many elderly women who stay there.
  When one of Gustave &#96s lovers dies mysteriously, Gustave finds himself the recipient
   of a priceless painting and the chief suspect in her murder.
            <p
              className="movie-card__director"
            >
              <strong>
                Director: 
                Wes Andreson
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: 
                Bill Murray, Edward Norton, Jude Law, Willem Dafoe
                 and other
              </strong>
            </p>
          </div>
          <div
            className="movie-card__text movie-card__row movie-tab movie-tab--details
        visually-hidden"
          >
            <div
              className="movie-card__text-col"
            >
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Director
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  Wes Andreson
                </span>
              </p>
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Starring
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  Bill Murray
                   
                  <br />
                  Edward Norton
                   
                  <br />
                  Jude Law
                   
                  <br />
                  Willem Dafoe
                   
                  <br />
                  Saoirse Ronan
                   
                  <br />
                  Tony Revoloru
                   
                  <br />
                  Tilda Swinton
                   
                  <br />
                  Tom Wilkinson
                   
                  <br />
                  Owen Wilkinson
                   
                  <br />
                  Adrien Brody
                   
                  <br />
                  Ralph Fiennes
                   
                  <br />
                  Jeff Goldblum
                   
                  <br />
                </span>
              </p>
            </div>
            <div
              className="movie-card__text-col"
            >
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Run Time
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  1
                  h 
                  39
                  m
                </span>
              </p>
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Genre
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  Drama
                </span>
              </p>
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Released
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  2014
                </span>
              </p>
            </div>
          </div>
          <div
            className="movie-card__reviews movie-card__row movie-tab movie-tab--reviews
        visually-hidden"
          >
            <div
              className="movie-card__reviews-col"
            >
              <div
                className="review"
              >
                <blockquote
                  className="review__quote"
                >
                  <p
                    className="review__text"
                  >
                    Discerning travellers and Wes Anderson fans will luxuriate
       in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
                  </p>
                  <footer
                    className="review__details"
                  >
                    <cite
                      className="review__author"
                    >
                      Kate Muir
                    </cite>
                    <time
                      className="review__date"
                      dateTime="March 15, 2019"
                    >
                      March 15, 2019
                    </time>
                  </footer>
                </blockquote>
                <div
                  className="review__rating"
                >
                  8.9
                </div>
              </div>
              <br />
              <div
                className="review"
              >
                <blockquote
                  className="review__quote"
                >
                  <p
                    className="review__text"
                  >
                    Anderson's films are too precious for some, but for those
      of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.
                  </p>
                  <footer
                    className="review__details"
                  >
                    <cite
                      className="review__author"
                    >
                      Bill Goodykoontz
                    </cite>
                    <time
                      className="review__date"
                      dateTime="March 25, 2019"
                    >
                      March 25, 2019
                    </time>
                  </footer>
                </blockquote>
                <div
                  className="review__rating"
                >
                  8
                </div>
              </div>
              <br />
              <div
                className="review"
              >
                <blockquote
                  className="review__quote"
                >
                  <p
                    className="review__text"
                  >
                    I didn't find it amusing, and while I can appreciate the
      creativity, it's an hour and 40 minutes I wish I could take back.
                  </p>
                  <footer
                    className="review__details"
                  >
                    <cite
                      className="review__author"
                    >
                      Amanda Greever
                    </cite>
                    <time
                      className="review__date"
                      dateTime="March 05, 2019"
                    >
                      March 5, 2019
                    </time>
                  </footer>
                </blockquote>
                <div
                  className="review__rating"
                >
                  8
                </div>
              </div>
              <br />
              <div
                className="review"
              >
                <blockquote
                  className="review__quote"
                >
                  <p
                    className="review__text"
                  >
                    The mannered, madcap proceedings are often delightful,
      occasionally silly, and here and there, gruesome and/or heartbreaking.
                  </p>
                  <footer
                    className="review__details"
                  >
                    <cite
                      className="review__author"
                    >
                      Matthew Lickona
                    </cite>
                    <time
                      className="review__date"
                      dateTime="March 15, 2018"
                    >
                      March 15, 2018
                    </time>
                  </footer>
                </blockquote>
                <div
                  className="review__rating"
                >
                  7.2
                </div>
              </div>
              <br />
              <div
                className="review"
              >
                <blockquote
                  className="review__quote"
                >
                  <p
                    className="review__text"
                  >
                    It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.
                  </p>
                  <footer
                    className="review__details"
                  >
                    <cite
                      className="review__author"
                    >
                      Paula Fleri-Soler
                    </cite>
                    <time
                      className="review__date"
                      dateTime="March 15, 2019"
                    >
                      March 15, 2019
                    </time>
                  </footer>
                </blockquote>
                <div
                  className="review__rating"
                >
                  7.6
                </div>
              </div>
              <br />
              <div
                className="review"
              >
                <blockquote
                  className="review__quote"
                >
                  <p
                    className="review__text"
                  >
                    It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.
                  </p>
                  <footer
                    className="review__details"
                  >
                    <cite
                      className="review__author"
                    >
                      Paula Fleri-Soler
                    </cite>
                    <time
                      className="review__date"
                      dateTime="March 4, 2019"
                    >
                      March 4, 2019
                    </time>
                  </footer>
                </blockquote>
                <div
                  className="review__rating"
                >
                  7
                </div>
              </div>
              <br />
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      />
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="main.html"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
